### PR TITLE
Provide hooks for arbitrary mimetypes handling.

### PR DESCRIPTION
**Pull request by Carreau**
This should allow at some point inline images, an math in the terminal.

Screen Shot 2019-11-17 at 10 48 45
 
Screen Shot 2019-11-17 at 10 49 45

### CODE REVIEW

**Comment by Carreau** [comment]
@asmeurer I know you are a Terminal IPython user with many extensions. 
Thoughts ?

File: docs/source/config/shell_mimerenderer.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        ipython.display_formatter.formatters[mime].enabled = True
+        ipython.mime_renderers[mime] = handler
+
+    def load_ipython_extension(ipython):
+        register_mimerenderer(ipython, 'image/png', imcat)
+        register_mimerenderer(ipython, 'image/jpeg', imcat)
+        register_mimerenderer(ipython, 'text/latex', mathcat)
+
+This example only work for iterm2 on macOS and skip error handling for brevity.
+One could also invoke an external viewer with ``subporcess.run()`` and a
```
Comment (asmeurer) [inline docs/source/config/shell_mimerenderer.rst]: This is spelled wrong.

**Comment by asmeurer** [comment]
I don't use IPython, but if this supports iTerm2 I think it's great. My only comment would be to make the images be full retina resolution.

File: docs/source/config/shell_mimerenderer.rst (hunk 1 above)
Comment (Carreau) [reply docs/source/config/shell_mimerenderer.rst]: Oh thanks.

**Comment by Carreau** [comment]
> I don't use IPython, but if this supports iTerm2 I think it's great. My only comment would be to make the images be full retina resolution.

Sorry I thought you had some custom prompt extensions at some points. I might be confusing with something else. Apologies for the wrong ping and thanks for spotting the typo.

**Comment by asmeurer** [comment]
I did have some custom stuff at one point but I gave up trying to finagle IPython at some point and just wrote my own thing https://github.com/asmeurer/mypython.

**Comment by Carreau** [comment]
> https://github.com/asmeurer/mypython.

Ah thanks, I'll try to remember this then.

**Pull request merged**